packages/bpk-component-modal/src/BpkModal.js

```diff
@@ -123,7 +123,7 @@ export const defaultProps = {
   closeOnEscPressed: true,
 };
 
-BpkModal.propTypes = { ...propTypes };
-BpkModal.defaultProps = { ...defaultProps };
+BpkModal.propTypes = { ...propTypes, onClose: PropTypes.func };
+BpkModal.defaultProps = { ...defaultProps, onClose: () => null };
```

Comment:
Rather than adding here you can probably move this to as looking at this it would be defining the onClose twice:

https://github.com/Skyscanner/backpack/pull/1937/files#diff-9b7c4907d7e12003dbda71f207de17f5R115